src/components/app.tsx

```diff
@@ -89,6 +103,60 @@ class AppBase extends React.PureComponent<Props, {}> {
     });
   }
 
+  private setSpec(spec: Object) {
+    const ajv = new Ajv({
+      validateSchema: true,
+      allErrors: true,
+      extendRefs: 'fail'
+    });
+    ajv.addMetaSchema(draft4Schemas, 'http://json-schema.org/draft-04/schema#');
+
+    const validateVl = ajv.compile(vlSchema);
+    const valid = validateVl(spec);
+
+    if (!valid) {
+      throw new Error("Invalid Schema:" + validateVl.errors.toString());
```

Comment:
Invalid spec?  (Schema is not invalid, but the spec does not conform to the schema?)